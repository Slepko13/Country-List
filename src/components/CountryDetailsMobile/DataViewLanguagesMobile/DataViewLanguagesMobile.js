import React from 'react';
import PropTypes from "prop-types";
import './DataViewLanguagesMobile.scss';

const DataViewLanguagesMobile = (props) => {

    let { title, data, position, background, picture } = props;
    if (!data) return <div
        className={position ?
            `DataViewLanguagesMobile ${position}` :
            "DataViewLanguagesMobile"}
        data-testid="position">
        <div className="absent">Data is absent today</div>
    </div>;


    return (
        <div
            className={position ?
                `DataViewLanguagesMobile ${position} ` :
                "DataViewLanguagesMobile"}
        >
            <img className="ellipse" src={picture} alt='ellipse' />
            <div className="innerVector" />

            <div className="info__title">{title}</div>
            <div className="info__content">
                {
                    data.length ?
                        data.map(item =>
                            <div
                                className="item"
                                key={item.name}
                                style={{ backgroundColor: background }}
                            >{item.name}</div>) :
                        <div className="item">n/a</div>
                }
            </div>
        </div>
    )
}

export default DataViewLanguagesMobile;

DataViewLanguagesMobile.propTypes = {
    title: PropTypes.string,
    data: PropTypes.array,
    position: PropTypes.string,
    background: PropTypes.string,
    picture: PropTypes.string
}