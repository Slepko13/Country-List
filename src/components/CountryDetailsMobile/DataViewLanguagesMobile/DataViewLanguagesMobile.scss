.DataViewLanguagesMobile {
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin: 0 0 25px 90px;

  .ellipse {
    position: absolute;
    top: 10px;
    right: 100%;
    margin: 0 25px 0 0;
  }

  .innerVector {
    position: absolute;
    width: 12px;
    height: 98%;
    background: url("../../../assets/images/Vector 10.png") center top;
    top: 38px;
    right: 100%;
    margin: 0 25px 0 0;
    transform: translateX(-50%);
  }

  .info__title {
    font-family: "Roboto";
    font-style: normal;
    font-weight: 300;
    font-size: 14px;
    line-height: 16px;
    text-align: center;
  }

  .info__content {
    font-family: "Roboto";
    font-style: normal;
    font-weight: bold;
    font-size: 20px;
    line-height: 24px;
    text-align: center;
    color: #313131;
    display: flex;
    justify-content: flex-start;
    flex-wrap: wrap;
    position: relative;


    .item {
      line-height: 23px;
      margin: 5px 10px 5px 0;
      border-radius: 18px;
      padding: 3px 10px;
    }

    .wrapper {
      position: relative;
      width: 70px;
    }
  }
}