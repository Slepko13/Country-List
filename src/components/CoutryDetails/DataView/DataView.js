import React from 'react';
import './DataView.scss';
import PropTypes from "prop-types";

const DataView = (props) => {
    let {title, data, position, addictions, maxWidth} = props;
    data = data === "" ? "n/a" : data === "Kiev" ? "Kyiv" : data;
    return (
        <div className={position ?
            `DataView ${position}` :
            "DataView"}
             data-testid="position"
        >
            <div className="info__title">{title}</div>
            <div className="info__content" style={{maxWidth}}>{data}<span style={{marginLeft: "10px"}}>{addictions}</span></div>
        </div>
    )
}
export default DataView;

DataView.propTypes = {
    title: PropTypes.string,
    data: PropTypes.string,
    position: PropTypes.string,
    addictions: PropTypes.string,
    maxWidth: PropTypes.string
}