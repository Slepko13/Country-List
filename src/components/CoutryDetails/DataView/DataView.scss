.DataView {
  margin-left: 20px;
  z-index: 5;
  @media screen and (max-width: 991px) {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    margin-bottom: 20px;

  }
    .info__title{
      font-family: "Roboto";
      font-style: normal;
      font-weight: 300;
      font-size: 14px;
      line-height: 16px;
      text-align: center;

    }
    .info__content {
      font-family: "Roboto";
      font-style: normal;
      font-weight: bold;
      text-align: center;
      color: #313131;
      line-height: 30px;
      font-size: 26px;
      max-height: 30px;
      text-overflow:ellipsis;
      overflow: hidden;
      max-width: 200px;

      &:hover {
        overflow: visible;
        max-width: 800px;
        background-color: #ffffff;
      }
    }
  }

