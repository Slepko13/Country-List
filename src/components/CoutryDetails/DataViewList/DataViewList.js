import React from 'react';
import './DataViewList.scss';

import SeeMore from "../SeeMore/SeeMore";


const DataViewList = (props) =>{

    let{title, data, position, addictions, direction} = props;
    let myData, restData;

    if(!data)    return  <div className={position ? `DataViewList + ${position} `: "DataViewList"}>
        <div className="absent">Data is absent today</div>
    </div>;

        if(data.length <= 3 && data.length > 0 ) {
            myData = [...data];
        }else if(data.length > 3) {
            myData = data.slice(0,3);
            restData = data.slice(3);
        }

    return (
        <div className={position ? `DataViewList + ${position} `: "DataViewList"}>
            <div className="info__title">{title}</div>
            <div className="info__content">{
                data.length ?
                    myData.filter(item=>item.name!=="null")
                        .map(item=>
                        <div
                            className="item"
                            key={item.name}
                        >{addictions}
                                <span className="item__symbol">{
                            !item.symbol ? null : item.symbol ==="null" ? "n/a": `"${item.symbol}"`
                                }</span>
                            {item.name.replace("UTC","GMT")}

                        </div>)
                :
                    <div className="item">n/a</div>
                    }
                {restData ?
                    <SeeMore
                        data={restData}
                        bottom=""
                        top="100%"
                        left=""
                        right=""
                    />: null}

            </div>
            {data.length > 3 ?<div style={{height:"20px"}}></div>:null}

        </div>
    )
}

export default DataViewList;