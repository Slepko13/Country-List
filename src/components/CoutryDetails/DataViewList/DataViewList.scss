.DataViewList{
  margin-right: 10px;
  position: relative;
  border : 2px solid red;

  .info__title{
    font-family: "Roboto";
    font-style: normal;
    font-weight: 300;
    font-size: 14px;
    line-height: 16px;
    text-align: center;
    margin-bottom: 20px;

  }
  .info__content {
    font-family: "Roboto";
    font-style: normal;
    font-weight: bold;
    text-align: center;
    color: #313131;
    display: flex;
    flex-direction: column;


    .item {
      text-align: left;
      line-height: 30px;
      font-size: 26px;
      max-height: 28px;
      text-overflow:ellipsis;
      overflow: hidden;
      max-width: 300px;
      &:hover {
        overflow: visible;
        max-width: 800px;
        background-color: #ffffff;
      }


      .item__symbol {
        margin-right: 10px;
      }
    }
  }
}