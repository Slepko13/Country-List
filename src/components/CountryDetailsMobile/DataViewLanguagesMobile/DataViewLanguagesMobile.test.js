import React from "react";
import DataViewLanguagesMobile from "./DataViewLanguagesMobile";
import DataViewList from "../DataViewListMobile/DataViewListMobile";
import {render} from "@testing-library/react";
import DataView from "../DataViewMobile/DataViewMobile";


describe("my DataViewLanguagesMobile component", () => {
    it("should renders with  additional classname depending on props ", () => {
        const { container, getByTestId } = render(
            <DataViewLanguagesMobile position="left" />
        );
        expect(getByTestId('position')).toHaveClass('left');
    });

    it("should render  without additional classname depending on props ", () => {
        const { container, getByTestId } = render(
            <DataViewLanguagesMobile position={null} />
        );
        expect(container.firstChild.classList.contains('left')).toBe(false)
    });

    describe("data is empty array", ()=>{
        const props = {
            data :[],
            title :"Languages",
            position: "left"
        }
        const component = shallow(<DataViewLanguagesMobile {...props}/>);
        it("should renders 'n/a ", () => {
            expect(component.find('.item').text()).toEqual('n/a');
        });
    });
    describe("data is absent", ()=>{
        const props = {
            title :"Languages",
            position: "left"
        }
        const component = shallow(<DataViewLanguagesMobile {...props}/>);
        it("should renders 'Data is absent today ", () => {
            expect(component.find('.absent').text()).toEqual('Data is absent today');
        });
    });
    describe("position is absent", ()=>{
        const props = {
            data : [
                {name :"english" },
                {name :"french" },
            ],
            title :"Languages",
        }
        const component = shallow(<DataViewLanguagesMobile {...props}/>);
        it("should renders correct className", () => {
            expect(component.find('.DataViewLanguagesMobile')).toHaveLength(1);
        });
    });

    describe("data length <= 3", ()=>{
        const props = {
            data : [
                {name :"english" },
                {name :"french" },

            ],
            title :"Languages",
            position: "left"
        }
        const component = shallow(<DataViewLanguagesMobile {...props}/>);
        it("should renders 2 items ", () => {
            expect(component.find('div.item')).toHaveLength(2);
        });

        it("should renders properly ", () => {
            expect(component).toMatchSnapshot();
        });

    });

    describe("data length > 3", ()=>{
        const props = {
            data : [
                {name :"english" },
                {name :"french" },
                {name :"german" },
                {name :"polish" }
            ] ,
            title :"Languages",
            position: "left"
        }
        const component = shallow(<DataViewLanguagesMobile {...props}/>);

        it("should renders properly ", () => {
            expect(component).toMatchSnapshot();
        });

        it("should renders 4 items  ", () => {
            expect(component.find('div.item')).toHaveLength(4);

        });

    });

});

