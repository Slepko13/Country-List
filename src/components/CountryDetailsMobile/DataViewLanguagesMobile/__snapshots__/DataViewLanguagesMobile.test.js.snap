// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my DataViewLanguagesMobile component data length <= 3 should renders properly  1`] = `
<div
  className="DataViewLanguagesMobile left "
>
  <img
    alt="ellipse"
    className="ellipse"
  />
  <div
    className="innerVector"
  />
  <div
    className="info__title"
  >
    Languages
  </div>
  <div
    className="info__content"
  >
    <div
      className="item"
      key="english"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    >
      english
    </div>
    <div
      className="item"
      key="french"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    >
      french
    </div>
  </div>
</div>
`;

exports[`my DataViewLanguagesMobile component data length > 3 should renders properly  1`] = `
<div
  className="DataViewLanguagesMobile left "
>
  <img
    alt="ellipse"
    className="ellipse"
  />
  <div
    className="innerVector"
  />
  <div
    className="info__title"
  >
    Languages
  </div>
  <div
    className="info__content"
  >
    <div
      className="item"
      key="english"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    >
      english
    </div>
    <div
      className="item"
      key="french"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    >
      french
    </div>
    <div
      className="item"
      key="german"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    >
      german
    </div>
    <div
      className="item"
      key="polish"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    >
      polish
    </div>
  </div>
</div>
`;
